nengo/networks/assoc_mem.py

```diff
@@ -75,9 +75,12 @@ def __init__(self, input_vectors, output_vectors=None,  # noqa: C901
         if is_iterable(output_vectors):
             output_vectors = np.matrix(output_vectors)
 
-        # Fail if number of input items and number of output items don't
-        # match
-        if input_vectors.shape[0] != output_vectors.shape[0]:
+        if input_vectors.shape[0] <= 0:
+            # Fail if number of input items <= 0
+            raise ValueError('Number of input vectors cannot be <= 0.')
```

Comment:
That's a good point. I would recommend we never use matrices in Nengo because a) they're never necessary, and b) basic operators like `*` perform different operations for arrays and matrices, making things very confusing.